src/expr/src/scalar/func/impls/int32.rs

```diff
@@ -121,31 +121,31 @@ sqlfunc!(
     #[sqlname = "i32tooid"]
     #[preserves_uniqueness = true]
     fn cast_int32_to_oid(a: i32) -> Oid {
-        Oid(a)
+        Oid(u32::from_ne_bytes(a.to_ne_bytes()))
```

Comment:
Ah, I was actually going out of my way not to do that! Thanks very much for reminding me to make that clearer. These reinterpret casts to/from i32 are super sketchy, and not something I think we ought to allow anywhere outside of where strictly required for PostgreSQL compatibility. (E.g., we shouldn't do the same sloppy casts when encoding OIDs in Avro .)

Point taken on the duplication, though! I reworked so that there are only three instances of the byte reinterpretation:

  * i32 -> oid
  * oid -> i32
  * text -> oid

Each of these is now marked with a comment explaining the situation.